keps/sig-apps/4563-evacuation-api/README.md

```diff
@@ -0,0 +1,1561 @@
+If the evacuation is no longer needed, the instigator should remove itself from the finalizers.
+The evacuation will be then deleted by the evacuation controller. In case the evacuator has set
+`.status.evacuationCancellationPolicy` to `Forbid`, the evacuation process cannot be cancelled, and
+the evacuation controller will wait to delete the pod until the pod has been terminated and removed
+from etcd.
+
+#### Evacuation Instigator Finalizer
+To distinguish between instigator and other finalizers, instigators should use finalizers in the
+following format: `evacuation.coordination.k8s.io/instigator_${EVACUATION_INSTIGATOR_SUBDOMAIN}`.
+For example, the node maintenance instigator would use the
+`evacuation.coordination.k8s.io/instigator_nodemaintenance.k8s.io` finalizer.
+
+Instigator finalizers with `evacuation.coordination.k8s.io/instigator_` prefix/qualified name prefix
+are automatically removed upon pod removal by the evacuation controller. If the instigator needs to
+perform additional final tasks before the Evacuation object is removed, it should create a second
+finalizer without this well-known prefix.
+
+The evacuation instigator should only add and remove its own finalizer and should not touch other
+finalizers with the `evacuation.coordination.k8s.io/` prefix.
+
+### Evacuator
+
+[Evacuee and Evacuator](#evacuee-and-evacuator) section provides a general overview.
+
+There can be multiple evacuators for a single pod, which can be given control of the evacuation by
+the evacuation controller.
+
+We can distinguish 4 different kinds of evacuators:
+- `partial`: it handles only a part of the evacuation process (e.g. releasing one kind of resource).
+  This evacuator should not terminate the pod.
+- `knowledgeable`: has more knowledge (e.g. specific to the deployed application) about how to
+  handle a more graceful evacuation than the controller of the pod (e.g. StatefulSet, Deployment).
+  This can result in the termination of the pod.
+- `controller`: handles full graceful evacuation in a generic way and should result in a pod
+  termination. It can be preempted by a `knowledgeable` evacuator.
+- `fallback`: can handle evacuation if all previous evacuators have failed. This means that
+  `knowledgeable` or `controller` evacuators have either reported no progress, or are simply not
+  implemented for the target pod.
+
+The evacuator should first mark the pods it is interested in evacuating (either partially or fully)
+with the `evacuation.coordination.k8s.io/priority_${EVACUATOR_CLASS}: ${PRIORITY}/${ROLE}`
+annotation. This annotation is parsed into the `Evacuator` type in the [Evacuation API](#evacuation-api).
+
+- `EVACUATOR_CLASS`: should be unique evacuator's DNS subdomain, the maximum length is 54
+  characters (`63 - len("priority_")`)
+- `PRIORITY` and `ROLE`
+  - `controller` should always set a `PRIORITY=10000` and `ROLE=controller`
+  - other evacuators should set `PRIORITY` according to their own needs (minimum value is 0,
```

Comment:
is 0 "highest priority" or "lowest priority"?  What does this priority govern?  Can two of them have the same prio?

Reply:
There are additional rules/restrictions/validations for the annotations now: https://github.com/kubernetes/enhancements/pull/4565/commits/45b9807fd845e596ca2116c98285360eda9caf81